### ISSUE TITLE
Setup type checking for backend

### ISSUE THREAD

**Issue by EddieTheCubeHead** [EddieTheCubeHead/Stagnum#24]
Ruff supports flake-8 type checking out of the box. Setup type checking. Use best judgement for choosing options and/or discuss with the rest of the group.

Acceptance criteria: ruff linter in CI/CD also does type checking

**Comment by tuomasmustakallio** [comment]
Blocked by #19

**Comment by EddieTheCubeHead** [comment]
Mypy doesn't support PEP 695 generics (yet). Postponed for now
